pkg/operator/baremetal_pod.go

```diff
@@ -17,13 +30,32 @@ var volumes = []corev1.Volume{
 			EmptyDir: &corev1.EmptyDirVolumeSource{},
 		},
 	},
+	{
+		Name: sharedSecret,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: secretName,
+				Items: []corev1.KeyToPath{
+					{
+						Key:  secretKey,
+						Path: "mariadb-password.json",
+					},
+				},
+			},
+		},
+	},
 }
 
 var volumeMounts = []corev1.VolumeMount{
 	{
 		Name:      sharedVolume,
 		MountPath: "/shared",
 	},
+	{
+		Name:      sharedSecret,
```

Comment:
I don't think this additional volume should be needed, we just need to pass the value as before via an environment variable.

Reply:
The maridb-password is already being passed into the relevant containers via environment variables. This is still part of the logic to add the Secret to the openshift-machine-api namespace so that it can later be used to update the env vars. I have a better way of doing it now.